<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>JGNsc • JGNsc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="JGNsc">
<meta property="og:description" content="JGNsc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">JGNsc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/JGNsc.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">References</a>
</li>
<li>
  <a href="../articles/web/JGNsc_data.html">Data</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../reference/index.html">
    <span class="fas fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/meichendong/JGNsc">
    <span class="fas fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="JGNsc_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>JGNsc</h1>
            
      
      
      <div class="hidden name"><code>JGNsc.Rmd</code></div>

    </div>

    
    
<div id="joint-gene-network-construction-by-single-cell-rna-sequencing-data-under-multiple-conditions" class="section level1">
<h1 class="hasAnchor">
<a href="#joint-gene-network-construction-by-single-cell-rna-sequencing-data-under-multiple-conditions" class="anchor"></a><em>Joint Gene Network Construction by Single Cell RNA Sequencing Data under Multiple Conditions</em>
</h1>
<p>Overview of JGNsc framework: <br><img src="Figureoverview.JPG" style="width:90.0%" alt="JGNsc Overview"></p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/">"devtools"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"meichendong/JGNsc"</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">JGNsc</span><span class="op">)</span></code></pre></div>
</div>
<div id="example-simulation" class="section level2">
<h2 class="hasAnchor">
<a href="#example-simulation" class="anchor"></a>Example – Simulation</h2>
<p>The following code simulates raw count data from more than one conditions.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>

<span class="co"># example: the first 20 genes have different structure, the rest of genes have the same structures.</span>
<span class="va">scenario</span> <span class="op">=</span> <span class="st">"DI20"</span> <span class="co"># can change this to DD/DI100/ID</span>
<span class="va">nsample</span> <span class="op">=</span> <span class="fl">100</span>

<span class="va">nivec.list.diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>nivec<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">10</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">20</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span>,nivec2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">20</span>,<span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="va">diffblk</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">)</span>
<span class="va">sigma.list.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generateSigmaList.html">generateSigmaList</a></span><span class="op">(</span>nivec.list <span class="op">=</span> <span class="va">nivec.list.diff</span>, structure <span class="op">=</span> <span class="st">"Diff S, Identical W"</span>, diffblk <span class="op">=</span> <span class="va">diffblk</span><span class="op">)</span>
<span class="co"># Create a list of count matrices</span>
<span class="va">countlist.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCountList.html">getCountList</a></span><span class="op">(</span>sigma.list <span class="op">=</span> <span class="va">sigma.list.1</span>, nvec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">nsample</span>,<span class="fl">2</span><span class="op">)</span>, a3<span class="op">=</span><span class="fl">2.5</span>, b3<span class="op">=</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>The network structure (simulation truth) is available at <code>countlist.1</code> above. Check the details by</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">countlist.1</span><span class="op">)</span>
<span class="va">observed.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">countlist.1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">countlist.1</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Run JGNsc imputation and continuization step, and the Joint Graphical Lasso (JGL) model with AIC as tuning parameter selection criterion. This may take 5 to 10 minutes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">simulation.result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/JGNsc.html">JGNsc</a></span><span class="op">(</span>observed.list <span class="op">=</span> <span class="va">observed.list</span>, min.cell <span class="op">=</span> <span class="fl">10</span>, runNetwork <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="co"># check the structure of the result</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">simulation.result</span><span class="op">)</span>
<span class="co"># check the AIC values</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">simulation.result</span><span class="op">$</span><span class="va">aic.table</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">V1</span>, y<span class="op">=</span><span class="va">V3</span>, color <span class="op">=</span> <span class="va">V2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"lambda1"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"AIC"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"lambda2"</span><span class="op">)</span></code></pre></div>
<p>Visualize the estimated partial correlation matrix. In this example, the sample size is the same as the number of genes, hence the estimated matrix is more noisy.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span>
<span class="co"># ESTIMATED</span>
<span class="va">partcorr.est1.trunc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/trunc_precision.html">trunc_precision</a></span><span class="op">(</span><span class="va">simulation.result</span><span class="op">$</span><span class="va">partcorr</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, threshold <span class="op">=</span> <span class="fl">0.0001</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="va">partcorr.est1.trunc</span>, cluster_rows <span class="op">=</span> <span class="cn">F</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="co"># SIMULATION TRUTH</span>
<span class="va">partcorr.true1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prec2partialcorr.html">prec2partialcorr</a></span><span class="op">(</span><span class="va">countlist.1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">precision</span><span class="op">)</span>
<span class="va">partcorr.true1.trunc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/trunc_precision.html">trunc_precision</a></span><span class="op">(</span><span class="va">partcorr.true1</span>, threshold <span class="op">=</span> <span class="fl">0.0001</span><span class="op">)</span>
<span class="co"># partcorr.true1[1:6,1:6]</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="va">partcorr.true1</span>, cluster_rows <span class="op">=</span> <span class="cn">F</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
</div>
<div id="example-mb-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-mb-data" class="anchor"></a>Example – MB data</h2>
<p>The data is from the <a href="https://www.nature.com/articles/s41586-019-1434-6">paper Hovestadt et al.</a>.<br>
The data is preprocessed by <a href="https://satijalab.org/seurat/">Seurat</a>.<br>
The processed example data is available upon request due to the data size.</p>
<div id="the-most-part-of-example-below-was-submitted-to-cluster--the-running-time-may-be-long-hours-" class="section level4">
<h4 class="hasAnchor">
<a href="#the-most-part-of-example-below-was-submitted-to-cluster--the-running-time-may-be-long-hours-" class="anchor"></a>The most part of example below was submitted to cluster. The running time may be long (hours).</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># setwd("your/data/path")</span>
<span class="co"># Seurat object</span>
<span class="va">sobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"H0102_seuratobj_3d.rds"</span><span class="op">)</span>  
<span class="co"># metabolism enzyme only pathway</span>
<span class="va">meta2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"Mammalian_Metabolic_enzyme_genes.csv"</span><span class="op">)</span></code></pre></div>
<p>Check demographic information by groups, for the gene of interest MYC:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getGENEbox.html">getGENEbox</a></span><span class="op">(</span>countmat <span class="op">=</span> <span class="va">sobj</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span>, group <span class="op">=</span> <span class="va">sobj</span><span class="op">$</span><span class="va">group3</span>, gene <span class="op">=</span> <span class="st">"MYC"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="J0001_MB_getGENEbox.png" alt=""><p class="caption">MB demographics MYC boxplot</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getGENEhist.html">getGENEhist</a></span><span class="op">(</span>countmat <span class="op">=</span> <span class="va">sobj</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span>, group <span class="op">=</span> <span class="va">sobj</span><span class="op">$</span><span class="va">group3</span>, gene <span class="op">=</span> <span class="st">"MYC"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="J0001_MB_getGENEhist.png" alt=""><p class="caption">MB demographics MYC histogram</p>
</div>
<p>Create a list of raw count matrices:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MB.3cond</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getObservedList.html">getObservedList</a></span><span class="op">(</span>mtx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">sobj</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span>, group <span class="op">=</span> <span class="va">sobj</span><span class="op">$</span><span class="va">group3</span>, geneSet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="va">meta2</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">)</span>, <span class="st">"MYC"</span>, <span class="st">"OTX2"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># the list of observed count matrices can be import as below</span>
<span class="co"># MB.3cond &lt;- readRDS("J0001_MB3cond.rds")</span></code></pre></div>
<p>Run JGNsc imputation and continuization step. The current version may take hours or longer if the data dimension is high. Faster algorithm is in development.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MB.3cond.continuous</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/JGNsc.html">JGNsc</a></span><span class="op">(</span>observed.list <span class="op">=</span> <span class="va">MB.3cond</span>, min.cell <span class="op">=</span> <span class="fl">10</span>, runNetwork <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<p>Tuning parameter selection by AIC, for JGL model:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getJGLTuningParamResult.html">getJGLTuningParamResult</a></span><span class="op">(</span>GauList <span class="op">=</span> <span class="va">MB.3cond.continuous</span><span class="op">$</span><span class="va">theta.star.npn</span><span class="op">)</span>
<span class="co"># transform the precision matrix to partial correlation</span>
<span class="va">partcorr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">jgl.res</span>, <span class="va">prec2partialcorr</span><span class="op">)</span>
<span class="co"># you can read in the calculated partial correlation matrices for the visualization steps</span>
<span class="co"># partcorr &lt;- readRDS("J0001_MB_partcorr.rds")</span></code></pre></div>
</div>
<div id="gene-set-enrichment-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-set-enrichment-analysis" class="anchor"></a>Gene Set Enrichment analysis</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Map2Pathways.html">Map2Pathways</a></span><span class="op">(</span>partcorr.list <span class="op">=</span> <span class="va">partcorr</span>,
                     conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">sobj</span><span class="op">$</span><span class="va">group3</span><span class="op">)</span>,
                     GeneInterest <span class="op">=</span> <span class="st">"MYC"</span>,
                     pathwayRef <span class="op">=</span> <span class="va">meta2</span>,
                     pathwayRef_geneVariable <span class="op">=</span> <span class="st">"GeneSymbol"</span>,
                     pathwayRef_pathVariable <span class="op">=</span> <span class="st">"SigmaMiniMap.Term"</span>,
                     threshold <span class="op">=</span><span class="fl">0</span><span class="op">)</span>
<span class="va">GSEA.table</span> <span class="op">&lt;-</span> <span class="va">test</span><span class="op">$</span><span class="va">GSEA.table</span></code></pre></div>
</div>
<div id="visualize-joint-networks-for-myc-connected-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-joint-networks-for-myc-connected-genes" class="anchor"></a>Visualize joint networks for MYC connected genes</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene1</span> <span class="op">=</span> <span class="st">"MYC"</span>
<span class="va">gconnect</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="va">test</span><span class="op">$</span><span class="va">cond.connect</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">)</span>, <span class="va">gene1</span><span class="op">)</span>
<span class="va">highlight</span> <span class="op">=</span> <span class="va">gconnect</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gene1</span><span class="op">)</span>
<span class="va">net1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_onenet.html">plot_onenet</a></span><span class="op">(</span><span class="va">partcorr</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">gconnect</span>,<span class="va">gconnect</span><span class="op">]</span>, gname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">sobj</span><span class="op">$</span><span class="va">group3</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, circlenet <span class="op">=</span> <span class="cn">T</span>, nodecolor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"orange"</span>,<span class="st">"lightblue"</span><span class="op">)</span><span class="op">[</span><span class="va">highlight</span><span class="op">+</span><span class="fl">1</span><span class="op">]</span>, family.vec <span class="op">=</span> <span class="va">highlight</span><span class="op">)</span>

<span class="va">gconnect</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="va">test</span><span class="op">$</span><span class="va">cond.connect</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">)</span>, <span class="va">gene1</span><span class="op">)</span>
<span class="va">highlight</span> <span class="op">=</span> <span class="va">gconnect</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gene1</span><span class="op">)</span>
<span class="va">net2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_onenet.html">plot_onenet</a></span><span class="op">(</span><span class="va">partcorr</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">gconnect</span>,<span class="va">gconnect</span><span class="op">]</span>, gname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">sobj</span><span class="op">$</span><span class="va">group3</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, circlenet <span class="op">=</span> <span class="cn">T</span>, nodecolor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"orange"</span>,<span class="st">"lightblue"</span><span class="op">)</span><span class="op">[</span><span class="va">highlight</span><span class="op">+</span><span class="fl">1</span><span class="op">]</span>, family.vec <span class="op">=</span> <span class="va">highlight</span><span class="op">)</span>

<span class="va">gconnect</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="va">test</span><span class="op">$</span><span class="va">cond.connect</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">)</span>, <span class="va">gene1</span><span class="op">)</span>
<span class="va">highlight</span> <span class="op">=</span> <span class="va">gconnect</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gene1</span><span class="op">)</span>
<span class="va">net3</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_onenet.html">plot_onenet</a></span><span class="op">(</span><span class="va">partcorr</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">gconnect</span>,<span class="va">gconnect</span><span class="op">]</span>, gname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">sobj</span><span class="op">$</span><span class="va">group3</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, circlenet <span class="op">=</span> <span class="cn">T</span>, nodecolor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"orange"</span>,<span class="st">"lightblue"</span><span class="op">)</span><span class="op">[</span><span class="va">highlight</span><span class="op">+</span><span class="fl">1</span><span class="op">]</span>, family.vec <span class="op">=</span> <span class="va">highlight</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">net3</span>, <span class="va">net1</span>, <span class="va">net2</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="J0001_MB_PLOTONENET.png" alt=""><p class="caption">MB Joint Networks</p>
</div>
</div>
</div>
<div id="example-gbm-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-gbm-data" class="anchor"></a>Example – GBM data</h2>
<p>The data is from the <a href="https://www.sciencedirect.com/science/article/pii/S0092867419306877?via%3Dihub">paper Neftel et al.</a>. The processed example data is available upon request due to the data size.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"N0102_seuratobj_3d_malignant_adult.rds"</span><span class="op">)</span>   </code></pre></div>
<p>The workflow is similar to the MB data example. Below we provide the list of count matrices of the four subtypes, and the list of estimated partial correlation matrices of the GBM data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GBM_4subtype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"J0001_GBM_ObservedList.rds"</span><span class="op">)</span>
<span class="va">GBM_partcorr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"J0001_GBM_partcorr.rds"</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Meichen Dong.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
